src/main/java/uk/gov/companieshouse/overseasentitiesapi/service/FilingsService.java

```diff
@@ -127,17 +132,21 @@ private void setFilingApiData(
 
     if (submissionDto.isForUpdate()) {
       getPublicAndPrivateData(submissionDto.getEntityNumber(), passThroughTokenHeader);
-      //Call a method to populate UpdateSubmission
-      filing.setKind(FILING_KIND_OVERSEAS_ENTITY_UPDATE);
       var publicPrivateDataCombiner = new PublicPrivateDataCombiner(
-          publicDataRetrievalService, privateDataRetrievalService, salt);
+              publicDataRetrievalService, privateDataRetrievalService, salt);
+
+      var updateSubmission = new UpdateSubmission();
+
+      updateSubmission.setEntityNumber(submissionDto.getEntityNumber());
+      updateSubmission.getChanges().addAll(overseasEntityChangeService.collateOverseasEntityChanges(
```

Comment:
I don't think we have a test verifying `overseasEntityChangeService.collateOverseasEntityChanges` is called and the setting of the entity number for an update submission, can you please check?

Reply:
Good spot, although i've added the necessary checks to the commented out test testFilingGenerationWhenSuccessfulWithoutTrustsAndWithIdentityChecksForUpdate in FilingServiceTest.java . This test should be uncommented and working pending completion of: https://companieshouse.atlassian.net/browse/UAR-663

Reply:
Thank you for updating this